diag_manager/fms_diag_yaml.F90

```diff
@@ -696,34 +692,61 @@ function get_total_num_vars(diag_yaml_id, diag_file_id) &
   end do
 end function
 
-!> @brief This checks if the file frequency and file frequency units in a diag file are valid and
-!! sets the integer equivalent
-subroutine set_file_freq(fileobj, file_freq, file_frequnit)
-  type(diagYamlFiles_type), intent(inout) :: fileobj         !< diagYamlFiles_type obj to check
-  character(len=*),         intent(in)    :: file_freq       !< File_freq as it is read from the diag_table
-  character(len=*),         intent(in)    :: file_frequnit   !< File_freq_units as it is read from the diag_table
-
-  integer           :: i                         !< For do loops
-  character(len=10) :: file_freq_units(MAX_FREQ) !< Array of file frequencies as a string
-  integer           :: err_unit                  !< Dummy error unit
-
-  file_freq_units = ""
-  read(file_freq, *, iostat=err_unit) fileobj%file_freq
-  read(file_frequnit, *, iostat=err_unit) file_freq_units
-
-  do i = 1, MAX_FREQ
-    if (fileobj%file_freq(i) >= -1) then
-      if (trim(file_freq_units(i)) .eq. "") &
-        call mpp_error(FATAL, "file_freq_units is required. &
-        &Check your entry for file:"//trim(fileobj%file_fname))
-
-      fileobj%file_frequnit(i) = set_valid_time_units(file_freq_units(i), &
-      "file_freq_units for file:"//trim(fileobj%file_fname))
-    else
-      return
+!> @brief This parses the freq, new_file_freq, or file_duration keys which are read in as a comma list
+subroutine parse_key(filename, buffer, file_freq, file_frequnit, var)
```

Comment:
Why isn't this done on the C side where all of the parsing is done?

Reply:
This parsing is simple and can just be done in fortran. I don't think there is any benefit in doing it in c.